src/main/java/de/tum/in/test/api/util/ClassMemberAccessor.java

```diff
@@ -0,0 +1,213 @@
+package de.tum.in.test.api.util;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+import java.util.Arrays;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+/**
+ * Provides utility methods to search for declared and inherited public,
+ * protected, and (package) private members of classes.
+ */
+class ClassMemberAccessor {
+	private ClassMemberAccessor() {
+	}
+
+	/**
+	 * Retrieve a method with arguments of a given class by its name.
+	 * <p>
+	 * Also recursively searches for an inherited method in all superclasses and
+	 * implemented interfaces.
+	 *
+	 * @param declaringClass The class that declares or inherits the method.
+	 * @param methodName     The name of the method.
+	 * @param findPrivate    True, if this method should search for (package)
+	 *                       private or protected (inherited) methods.
+	 * @param parameterTypes The parameter types of this method.
+	 * @return The wanted method.
+	 * @throws NoSuchMethodException Thrown if the specified method cannot be found.
+	 */
+	static Method getMethod(Class<?> declaringClass, String methodName, boolean findPrivate, Class<?>[] parameterTypes)
```

Comment:
I'd change the name of "declaringClass" to something like clazz, since it does not actually have to declare the method.
And findPrivate might be changed as well, e.g. findNonPublic

Reply:
Though of other alternatives, but non-public and restricted were the only names I came up with right now.

Reply:
768b890

I changed both `declaringClass` → `clazz` and `findPrivate` → `findNonPublic` here and for other occurrences.

To stay consistent, the user-facing methods of the `ReflectionTestUtils` are now named with `NonPublic` as well (`invokeNonPublicMethod`, …).